schemas/oic.r.sp.json

```diff
@@ -8,19 +8,14 @@
       "properties": {
         "supportedprofiles": {
           "type": "array",
-          "description": "Array of supported security profiles",
+          "description": "Array of supported Security Profiles",
           "items": {
-            "$ref": "https://openconnectivityfoundation.github.io/security/schemas/oic.sec.sp.name.json#/definitions/oic.sec.sp.name"
+             "type": "string"
           }
         },
         "currentprofile": {
-          "description": "Security profile currently active",
-          "$ref": "https://openconnectivityfoundation.github.io/security/schemas/oic.sec.sp.name.json#/definitions/oic.sec.sp.name"
-        },
-        "credid":  {
-          "type": "integer",
-          "description": "A value is returned if the security profile defines its usage. When defined, credid Property refers to the /oic/sec/cred Resource entry where the credential is used to authorize the available security profiles (e.g. OCF Rooted Certificate Chain)",
-          "readOnly": true
+          "description": "Security Profile currently active",
+          "type": "string"
```

Comment:
Should we create a separate type (oic.sec.ocfSecurityProfileOIDtype) with validation as enumeration or some regular expression?

Reply:
I think that would be a good idea.  Maybe ask Jeremy what he things.

Reply:
I wouldn't want people to confuse the enumerations for the string values, but if that's not an issue --- this naming convention would fall inline with other security enums.  The spec shouldn't move faster than the codebase, so enumeration should be ok vs. regex.

Reply:
When I said "I think that would be a good idea" I was specifically referring to the use of enumerated values, not a regex pattern.
However, thinking about it more, we could set the regex pattern now and it will never change since the first 15 digits or so never change and that digit (or two) will always follow the same format. The benefit of this is that we won't have to always updated the schema as new versions of the profiles are added.
Note that we will be checking the exact value of the "currentprofile" as part of certification testing so we don't HAVE to force it to be specific values by using an enum.
So my current vote (if I have one) is to define this as a string with a regex pattern.